src/test/parallel.py

```diff
@@ -0,0 +1,864 @@
+  parser.add_option('--dump_json_test_results', type='string', default=None,
+                    help='Saves the results of the tests as a JSON machine-'
+                         'readable file. The format of the file is specified at '
+                         'https://www.chromium.org/developers/the-json-test-results-format')
+  parser.add_option('--timeout', type='int', default=None,
+                    help='Interrupt all remaining processes after the given '
+                         'time (in seconds).')
+  parser.add_option('--serialize_test_cases', action='store_true',
+                    default=False, help='Do not run tests from the same test '
+                                        'case in parallel.')
+# Bitcoin Core specific options parsing: start
+  parser.add_option('-l', '--write_list', action='store_true',
+                    default=False, help='Write test list from the binary to '
+                                        '{} and exit.'.format(file_test_list))
+  parser.add_option('-b', '--binary', type=str, default=default_binary,
+                    help='The test binary (default: {})'.format(default_binary))
+# Bitcoin Core specific options parsing: end
+  return parser
+
+
+def main():
+  # Remove additional arguments (anything after --).
+  additional_args = []
+
+  for i in range(len(sys.argv)):
+    if sys.argv[i] == '--':
+      additional_args = sys.argv[i+1:]
+      sys.argv = sys.argv[:i]
+      break
+# Bitcoin Core specific options parsing: start
+  config = configparser.ConfigParser()
+  configfile = os.path.abspath(os.path.dirname(__file__)) + "/../../test/config.ini"
+  config.read_file(open(configfile))
+  env = config['environment']
+  FILE_TEST_LIST = os.path.join(env['SRCDIR'], 'src', 'test', 'test_list.txt')
+  DEFAULT_BINARY = os.path.join(env['BUILDDIR'],'src', 'test','test_bitcoin'+env['EXEEXT'])
+  parser = default_options_parser(DEFAULT_BINARY, FILE_TEST_LIST)
+  (options, binaries) = parser.parse_args()
+  if options.write_list:
+      write_test_list(options.binary, FILE_TEST_LIST)
+      sys.exit(0)
+  test_list = read_test_list(FILE_TEST_LIST)
+  binaries = [options.binary]
+# Bitcoin Core specific options parsing: end
+
+  if (options.output_dir is not None and
+      not os.path.isdir(options.output_dir)):
+    parser.error('--output_dir value must be an existing directory, '
+                 'current value is "%s"' % options.output_dir)
+
+  # Append gtest-parallel-logs to log output, this is to avoid deleting user
+  # data if an user passes a directory where files are already present. If a
+  # user specifies --output_dir=Docs/, we'll create Docs/gtest-parallel-logs
+  # and clean that directory out on startup, instead of nuking Docs/.
+  if options.output_dir:
+    options.output_dir = os.path.join(options.output_dir,
+                                      'gtest-parallel-logs')
+
+  if binaries == []:
+    parser.print_usage()
+    sys.exit(1)
+
+  if options.shard_count < 1:
```

Comment:
Nit: if you are going to verify the input is valid, you can check some other fields as well

Right now you could input a negative for `workers`, `repeat`, `timeout`, etc. without complaint